app/src/main/java/dev/arkbuilders/arkmemo/ui/viewmodels/GraphicNotesViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package dev.arkbuilders.arkmemo.ui.viewmodels
+
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Path
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dev.arkbuilders.arkmemo.models.GraphicNote
+import dev.arkbuilders.arkmemo.graphics.SVG
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import java.util.Stack
+import javax.inject.Inject
+
+@HiltViewModel
+class GraphicNotesViewModel @Inject constructor(): ViewModel() {
+
+    private val _notes = MutableStateFlow(listOf<GraphicNote>())
+    val notes: StateFlow<List<GraphicNote>> = _notes
+    private var paintColor = Color.BLACK
+
+    private var strokeWidth = 10f
+
+    val paint get() = Paint().also {
+        it.color = paintColor
+        it.style = Paint.Style.STROKE
+        it.strokeWidth = strokeWidth
+        it.strokeCap = Paint.Cap.ROUND
+        it.strokeJoin = Paint.Join.ROUND
+        it.isAntiAlias = true
+    }
+
+    private val editPaths = Stack<DrawPath>()
+
+    private var svg = SVG()
+
+    fun onNoteOpened(note: GraphicNote) {
+        viewModelScope.launch {
+            if (editPaths.isNotEmpty()) editPaths.clear()
+            editPaths.addAll(note.svg?.getPaths()!!)
+            svg = note.svg.copy()
```

Comment:
Do we really need `copy()`? Can't we just use `note.svg` value?

Reply:
Yes we need a copy to avoid mutating the old resource, this is for version tracking sake. Remember in version tracking we create new resource after every edit. We don't delete the old resource, we still keep it in resource history. I tested it before, after save the edits were in both new and old notes